app/tools/handlers.js

```diff
@@ -39,3 +45,15 @@ exports.restart = (req, res, next) => {
     })
     .catch(next);
 };
+
+
+exports.deploy = (req, res, next) => {
+  const { url_for } = require('../routes'); // eslint-disable-line global-require
+
+  new ToolDeployment({tool_name: req.params.name}).create()
+
+  req.session.flash_messages.push(`Deploying '${req.params.name}'...this may take up to 5 minutes`);
+  setTimeout(() => {
+    res.redirect(url_for('tools.list'));
+  }, 2000);
```

Comment:
This timeout is not the best solution - you can't guarantee that the deployment will even start within an arbitrary time. This should explicitly wait for some event, or not wait at all and the tool list page should update dynamically.

Just leaving this comment here for the record - we already agreed that this solution is ok for now.